blog/_src/posts/2017-06-06-spring-2017-pl-junior-retrospective.md

```diff
@@ -0,0 +1,162 @@
+deeper knowledge to the rest of the group, with the result being that
+the speaker has a better understanding and everyone else has about the
+same understanding compared to when everyone just reads the paper.
+
+One idea from last semester that we decided to keep is to spend a
+length of time (possibly an entire semester) on a topic rather than
+having a new topic each week. Staying on the same theme helps with
+retention as well as allowing for deeper investigation.
+
+In that spirit, we chose three themes for the semester: program
+synthesis, the sequent calculus, and logic programming. Mostly by
+chance, these topics have interesting connections to each other, and
+we even had several PL Grown-Up Seminars this semester on program
+synthesis!
+
+## Synthesis
+
+The first paper on program synthesis that we looked at was
+[A Deductive Approach to Program Synthesis][deductive] by Zohar and
+Manna. We chose this paper because it's old and has a lot of citations
+so it's probably Important. It was interesting and provided an OK
+introduction to proof search but the method presented seems far
+removed from modern synthesis techniques.
+
+The next paper was
+[Programmatic and Direct Manipulation, Together][progdirect] by Chugh
+et al which presents the [Sketch-n-Sketch][sketchnsketch] system.
+Sketch-n-Sketch is a cool system. It demonstrates that a narrow
+application of synthesis - trying to fill in the constant values in a
+program (sketching) - can be used for great effect. We were left
+wondering, however, if it was too narrow an application of synthesis
+to give much of an indication of what the entire field is like.
+
+We concluded our program synthesis segment with
+[Type-and-Example-Directed Program Synthesis][typeandexample] by Osera
+and Zdancewic, another relatively recent paper. This seems like a
+relevant paper because we are under the impression that using examples
+to do synthesis is a big thing right now. Using types to constrain the
+search is another interesting perspective on techniques for synthesis.
+
+While each of theses papers had merits, none was so comprehensive as
+to be a necessary inclusion in any future look at program synthesis
+for pl junior
+
+## Sequent Calculus
+
+We followed up the program synthesis unit with a week on the sequent
+calculus. The seminar presentation was based on a paper by
+[Herbelin][Herbelin]. [Gabriel’s thesis][gab] (chapter 4) includes
+maybe a more suitable modern introduction to the sequent calculus.
+
+It might have been better to do sequent calculus first because there
+is a modern branch of proof search based on the sequent calculus.
+Presenting this first would have allowed us to look into proof search
+for program synthesis.
+
+An additional problem is that it was insufficiently motivated. Either
+skipping the topic or spending more time on it would be preferable,
+since one week was just enough to describe the sequent calculus but
+not enough to apply it.
```

Comment:
Add "For this topic to be worthwhile, it would best be used as the basis for subsequent readings that directly reference it."